frontend/src/stimulus/controllers/dynamic/work-packages/activities-tab/index.controller.ts

```diff
@@ -0,0 +1,422 @@
+      setTimeout(() => {
+        scrollableContainer.scrollTop = scrollableContainer.scrollHeight;
+      }, 400);
+    }
+  }
+
+  setFilterToOnlyComments() { this.filterValue = 'only_comments'; }
+  setFilterToOnlyChanges() { this.filterValue = 'only_changes'; }
+  unsetFilter() { this.filterValue = ''; }
+
+  private getCkEditorInstance():ICKEditorInstance | null {
+    const AngularCkEditorElement = this.element.querySelector('opce-ckeditor-augmented-textarea');
+    return AngularCkEditorElement ? jQuery(AngularCkEditorElement).data('editor') as ICKEditorInstance : null;
+  }
+
+  private getInputContainer():HTMLElement | null {
+    return this.element.querySelector('#input-container');
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isMobile():boolean {
+    return window.innerWidth < 1279;
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isSmViewPort():boolean {
+    return window.innerWidth < 543;
+  }
+
+  // TODO: get rid of static width value and reach for a more CSS based solution
+  private isMdViewPort():boolean {
+    return window.innerWidth >= 543 && window.innerWidth < 1279;
+  }
+
+  private addEventListenersToCkEditorInstance() {
+    const editor = this.getCkEditorInstance();
+    if (editor) {
+      this.addKeydownListener(editor);
+      this.addKeyupListener(editor);
+      this.addBlurListener(editor);
+    }
+  }
+
+  private addKeydownListener(editor:ICKEditorInstance) {
+    editor.listenTo(
+      editor.editing.view.document,
+      'keydown',
+      (event, data) => {
+        // taken from op-ck-editor.component.ts
+        // eslint-disable-next-line @typescript-eslint/no-unsafe-enum-comparison
+        if ((data.ctrlKey || data.metaKey) && data.keyCode === KeyCodes.ENTER) {
+          void this.onSubmit();
+          event.stop();
+        }
+      },
+      { priority: 'highest' },
+    );
+  }
+
+  private addKeyupListener(editor:ICKEditorInstance) {
+    editor.listenTo(
+      editor.editing.view.document,
+      'keyup',
+      (event) => {
+        this.adjustJournalContainerMargin();
+        event.stop();
+      },
+      { priority: 'highest' },
+    );
+  }
+
+  private addBlurListener(editor:ICKEditorInstance) {
+    editor.listenTo(
+      editor.editing.view.document,
+      'change:isFocused',
+      () => {
+        // without the timeout `isFocused` is still true even if the editor was blurred
+        // current limitation:
+        // clicking on empty toolbar space and the somewhere else on the page does not trigger the blur anymore
+        setTimeout(() => {
+          // eslint-disable-next-line @typescript-eslint/no-unsafe-member-access
+          if (!editor.ui.focusTracker.isFocused) {
+            this.hideEditorIfEmpty();
+            if (this.isMobile()) { this.scrollInputContainerIntoView(300); }
+          }
+        }, 0);
+      },
+      { priority: 'highest' },
+    );
```

Comment:
Same as above, duplicating op-ckeditor code. I'd like to see this remain in the op-ckeditor custom element so it can be reused.